File: quiz_app_flutter/lib/domain/entities/result/result_data.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'result_data.freezed.dart';
part 'result_data.g.dart';

@freezed
class ResultItem with _$ResultItem {
  const factory ResultItem({
    required int questionId,
    required String question,
    required bool result,
    String? answerSectionTitle,
    required List<String> correctAnswer,
    required String explanation,
    @Default(false) bool isSkipped,
  }) = _ResultItem;

  factory ResultItem.fromJson(Map<String, dynamic> json) => _$ResultItemFromJson(json);
}

@freezed
class ResultData with _$ResultData {
  const factory ResultData({
    String? quizTitle,
    String? quizDescription,
    required List<ResultItem> resultItems,
    required int totalCorrectAnswers,
    required int totalQuestions,
    required int resultPercentage,
  }) = _ResultData;

  factory ResultData.fromJson(Map<String, dynamic> json) => _$ResultDataFromJson(json);
}

File: quiz_app_kmp/composeApp/src/commonMain/kotlin/com/tnm/quizappkmp/domain/model/result/ResultData.kt
package com.tnm.quizappkmp.domain.model.result

import kotlinx.serialization.Serializable

@Serializable
data class ResultData(
    val quizTitle: String?,
    val quizDescription: String? = null,
    val resultItems: List<Item>,
    val totalCorrectAnswers: Int,
    val totalQuestions: Int,
    val resultPercentage: Int,
) {
    @Serializable
    data class Item(
        val questionId: Int,
        val question: String,
        val result: Boolean,
        val answerSectionTitle: String? = null,
        val correctAnswer: List<String>,
        val explanation: String,
        val isSkipped: Boolean = false
    )
}
